### PR TITLE
Add trimmed fields in SMRRecord

### PR DESCRIPTION
**Pull request by lixinchengdu**
## Overview

Add trimmed fields in SMRRecord to indicate being trimmed by LogUnit servers. 

## Checklist (Definition of Done):

- [x] There are no TODOs left in the code
- [x] [Coding conventions](https://github.com/CorfuDB/CorfuDB/wiki/Corfu-Style-Guidelines) (e.g. for logging, unit tests) have been followed
- [x] Change is covered by automated tests
- [x] Public API has Javadoc

### CODE REVIEW

File: runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiSMREntry.java [hunk 1]
```diff
@@ -81,6 +89,10 @@ public void setEntry(ILogData entry) {
 
     @Override
     public List<SMREntry> getSMRUpdates(UUID id) {
+        if (trimmed) {
+            return Collections.singletonList(SMREntry.TRIMMED_ENTRY);
```
Comment (WenbinZhu) [inline runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiSMREntry.java]: I think we should return empty list here.

Comment (lixinchengdu) [reply runtime/src/main/java/org/corfudb/protocols/logprotocol/MultiSMREntry.java]: done.

**Comment by annym** [comment]
@lixinchengdu can you please rebase this, so we can start reviewing.

**Comment by lixinchengdu** [comment]
> @lixinchengdu can you please rebase this, so we can start reviewing.

Thanks very much for the reminder. I will rebase it as soon as possible.

**Pull request merged**